app/controllers/test_reports_controller.rb

```diff
@@ -61,6 +66,17 @@ def child_loop(job_tree, job_id, indent_num)
     job_tree[job_id][:children].reverse_each { |child_job_id| child_loop(job_tree, child_job_id, indent_num) }
   end
 
+  def ganerate_pagenation(search_word = nil)
+    if search_word.nil?
+      Kaminari.paginate_array(Job.all_root_jobs.to_a, total_count: Job.all_root_jobs.length).page(params[:page]).per(PER_PAGE)
+    else
+      Kaminari.paginate_array(
+        Job.all_root_jobs.searched_root_jobs(params[:search_word]).to_a,
+        total_count: Job.all_root_jobs.searched_root_jobs(params[:search_word]).length
+      ).page(params[:page]).per(PER_PAGE)
+    end
```

Comment:
I think there's a little bit of duplicate parts.

How about fixing it like this:
```ruby
          elements = if search_word.nil?
                        Job.all_root_jobs.to_a
                      else
                        Job.all_root_jobs.searched_root_jobs(params[:search_word]).to_a
                      end
          Kaminari.paginate_array(elements, total_count: elements.length).page(params[:page]).per(PER_PAGE)
```

Reply:
i agree your suggestion